src/wandbot/apps/ZD/ZDWandbot.py

```diff
@@ -0,0 +1,116 @@
+
+    def create_new_ticket(self, questionText):
+        self.zenpy_client.tickets.create(Ticket(subject="WandbotTest4", description=questionText, status = 'new', priority = 'low', tags=["botTest"]))
+
+    def fetch_new_tickets(self):
+        new_tickets = self.zenpy_client.search(type='ticket', status='new', group_id="360016040851")
+        # Filtering based on specific requirements
+        filtered_tickets = [ticket for ticket in new_tickets if 'forum' in ticket.tags]
+        # filtered_tickets = [ticket for ticket in new_tickets if 'bottest' in ticket.tags]         # for testing purposes only
+        filtered_ticketsNotAnswered = [ticket for ticket in filtered_tickets if 'answered_by_bot' not in ticket.tags]         # for testing purposes only
+
+        return filtered_ticketsNotAnswered
+
+    def extract_question(self, ticket):
+        description = ticket.description
+        # Preprocessing
+        question = description.lower().replace('\n', ' ').replace('\r', '')
+        question = question.replace('[discourse post]','')
+        question = question[:4095]
+
+        return question
+    
+    async def generate_response(self, question, ticket_id):
+        try:
+            chat_history = []
+
+            response = self.api_client.query(question=question, chat_history=[])
+            if response == None:
+                raise Exception("Recieved no response") 
+
+        except Exception as e:
+            logger.error(f"Error: {e}")
+            response = 'Something went wrong!'
+            return response
+    
+        return response.answer
+
+    #TODO: add the necessary format we want to depending on ticket type
+    def format_response(self, response):
+        response = str(response)
+        max_length = 2000
+        if len(response) > max_length:
+            response = response[:max_length] + '...'
+        return response+"\n\n-WandBot 🤖"
+
+    def update_ticket(self, ticket, response):
+        try:
+            comment = Comment(body=response)
+            ticket.comment = Comment(body=response, public=False)
+
+            ticket.status="open"
+            ticket.tags.append('answered_by_bot')
+            self.zenpy_client.tickets.update(ticket)
+        except Exception as e:
+            logger.error(f"Error: {e}")
+
+    #TODO add feedback gathering
+    def gather_feedback(self, ticket):
+        try:
+            ticket.comment = Comment(body="How did we do?", public=False)
+            self.zenpy_client.tickets.update(ticket)
+        except Exception as e:
+            logger.error(f"Error: {e}")
+
+    async def main(self):
+        # test tickets
+        # self.create_new_ticket("How Do I start a run?")
+        # self.create_new_ticket("Is there a way to programatically list all projects for a given entity?")
+        while True:
+            await asyncio.sleep(120)
+
+            new_tickets = self.fetch_new_tickets()
+            logger.info(f"New unanswered Tickets: {new_tickets}")
```

Comment:
Maybe just log the number of unanswered tickets here. Logging a list of tickets with all their content can fill-up the logs making debugging harder.